internal/utils/comparators/deployment/comparator.go

```diff
@@ -0,0 +1,80 @@
+package deployment
+
+import (
+	"reflect"
+
+	log "github.com/ViaQ/logerr/v2/log/static"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	apps "k8s.io/api/apps/v1"
+)
+
+// AreSame compares deployments for equality and return true equal otherwise false
+func AreSame(current *apps.Deployment, desired *apps.Deployment) (bool, string) {
+
+	if !utils.PodVolumeEquivalent(current.Spec.Template.Spec.Volumes, desired.Spec.Template.Spec.Volumes) {
+		log.V(3).Info("deployment volumes change", "deploymentName", current.Name)
+		return false, "volumes"
+	}
+
+	if len(current.Spec.Template.Spec.Containers) != len(desired.Spec.Template.Spec.Containers) {
+		log.V(3).Info("Deployment number of containers changed", "DeploymentName", current.Name)
+		return false, "numberOfContainers"
+	}
+
+	if isDeploymentImageDifference(current, desired) {
+		log.V(3).Info("Deployment image change", "Deployment", current.Name)
+		return false, "image"
+	}
+
+	if utils.AreResourcesDifferent(current, desired) {
+		log.V(3).Info("Deployment resource(s) change", "DeploymentName", current.Name)
+		return false, "resources"
+	}
+
+	if !utils.EnvValueEqual(current.Spec.Template.Spec.Containers[0].Env, desired.Spec.Template.Spec.Containers[0].Env) {
```

Comment:
I'm surprised we did not spin through the containers here

Reply:
Changed to loop through all containers to compare